packages/cli/tests/e2e/bot/ProvisionAppServiceNotificationBot.dotnet.tests.ts

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @author Yefu Wang <yefuwang@microsoft.com>
+ */
+
+import { happyPathTest } from "./NotificationBotHappyPathCommon";
+
+happyPathTest("dotnet");
```

Comment:
It's great to do this kind of encapsulation - `happyPathTest`.
I'm confused how can `happyPathTest("dotnet")` handle both CommandResponseBot & NotificationBot?

Reply:
There are two implementations: [CommandBotHappyPathCommon.ts](https://github.com/OfficeDev/TeamsFx/pull/5233/files#diff-c6eb1c8c32ee559cb66d9b34c4b28eb33bca059a54b54b4e3c132e194410af1c) and [NotificationBotHappyPathCommon.ts](https://github.com/OfficeDev/TeamsFx/pull/5233/files#diff-1561561b68cf175a567f553f820ef20e188de4a59fb6e0da5eb9ce37e236e2f5)

Reply:
Got it.